### PR TITLE
fix(accounts-db): general bug/perf fixes

### PR DESCRIPTION
**Pull request by 0xNineteen**
see the first few comments (by myself) on the main code changes this pr introduces, the tldr is:
- fixing an over-alloc bug which was causing OOM errors
- fixing slow metdata reading
- removing random print statements (mb)
- making gossip service optional when loading/validating a snapshot
- more docs on debugging info
- fix double deinit on gossip service

### CODE REVIEW

File: src/accountsdb/snapshots.zig [hunk 1]
```diff
@@ -431,7 +431,13 @@ pub const SnapshotFields = struct {
...
         };
         defer file.close();
-        return try decodeFromBincode(allocator, file.reader());
+
+        const size = (try file.stat()).size;
+        const contents = try file.readToEndAllocOptions(allocator, size, size, @alignOf(u8), null);
+        defer allocator.free(contents);
+
+        var fbs = std.io.fixedBufferStream(contents);
+        return try decodeFromBincode(allocator, fbs.reader());
```
Comment (0xNineteen) [inline src/accountsdb/snapshots.zig:440]: this was the slow metadata reading fix

File: src/cmd/cmd.zig [hunk 2]
```diff
@@ -452,18 +481,49 @@ fn validator() !void {
...
+
+    const snapshot_result = try loadSnapshot(
+        allocator,
+        app_base.logger,
+        null,
+        true,
+        false,
+    );
+    defer snapshot_result.deinit();
+}
```
Comment (0xNineteen) [inline src/cmd/cmd.zig:500]: this is a new command which allows us to validate an existing snapshot in one command

File: docs/debugging.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+- https://gist.github.com/qileq/49fbeff99def200179001d551c0a7036
+- also need to increase the os max number of vsnodes: `sudo sysctl kern.maxvnodes=100100100`
+
+# using vscode with sig unit tests 
+
+move the two files in (`docs/files/`) to the `.vscode` folder in root: 
+- `launch.json`
+- `tasks.json`
+
+add breakpoints and everything should work as normal in vscode
```
Comment (0xNineteen) [inline docs/debugging.md:14]: thinking ideally we start to fill this doc up with useful things for debugging/getting setup

**Review by InKryption — APPROVED** [review]
This all generally LGTM. I think it'd be good to merge soon to get these fixes & functionality into the rest of our workflows - I'll leave that to your discretion.

**Pull request merged**